### PR TITLE
Remove _get_shape  from Trajectory base

### LINKED ISSUE #42: Get correct shape of k-space trajectory tensor
Currently, the KTrajectory ABC ([_KTrajectory.py](https://github.com/ckolbPTB/mrpro/blob/main/src/mrpro/data/_KTrajectory.py)) has a method `_get_shape()` to calculate the correct shape of the traj torch.tensor:

```python

class KTrajectory(ABC):
    """Base class for k-space trajectories."""

    @staticmethod
    def _get_shape(header: KHeader) -> tuple[int, ...]:
        """Get the shape of the trajectory for the given header."""
        limits = header.encoding_limits
        other_dim = np.prod(
            [
                getattr(limits, field.name).length
                for field in dataclasses.fields(limits)
                if field.name not in ('k0', 'k1', 'k2', 'segment')
            ]
        )
        shape = (
            other_dim,
            3,
            limits.k2.length,
            limits.k1.length,
            limits.k0.length,
        )
        return shape

```

A discussion came up if this still works for all edge cases like acquisitions with freq/phase oversampling, partial fourier etc.

I did some test measurements with a simple FLASH sequence that @hdillinger compiled with the following default settings:

- FOV 192 x 192 x 5 mm³
- matrix size 192 x 192 x 1

I converted the Siemens XA61 raw data files to ISMRMRD files using the latest siemens_to_ismrmrd converter, read in the data using the MRPro KData class and extracted the following infos from the corresponding KHeader objects:

```python
for header in header_list:
    print(f'name: {header.protocol_name} - enc matrix: {header.encoding_matrix} - recon matrix: {header.recon_matrix}')

name: FLASH_basic - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_phase_oversampling_20 - enc matrix: SpatialDimension(x=384, y=232, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_phase_resolution_75 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_phase_partialfourier_6_8 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_avg4 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_slices3 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_measures5 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
name: FLASH_GRAPPA2 - enc matrix: SpatialDimension(x=384, y=192, z=1) - recon matrix: SpatialDimension(x=192, y=192, z=1)
``` 

```python
for header in header_list:
    print(f'name: {header.protocol_name} - {header.encoding_limits.k1} (length = {header.encoding_limits.k1.length})')

name: FLASH_basic - Limits(min=0, max=191, center=96) (length = 192)
name: FLASH_phase_oversampling_20 - Limits(min=0, max=230, center=115) (length = 231)
name: FLASH_phase_resolution_75 - Limits(min=0, max=143, center=72) (length = 144)
name: FLASH_phase_partialfourier_6_8 - Limits(min=0, max=143, center=96) (length = 144)
name: FLASH_avg4 - Limits(min=0, max=191, center=96) (length = 192)
name: FLASH_slices3 - Limits(min=0, max=191, center=96) (length = 192)
name: FLASH_measures5 - Limits(min=0, max=191, center=96) (length = 192)
name: FLASH_GRAPPA2 - Limits(min=0, max=190, center=96) (length = 191)
```

```python
for kdata in kdata_list:
    print(f'name: {kdata.header.protocol_name} - "_get_shape" return: {DummyTrajectory()._get_shape(kdata.header)}')  

name: FLASH_basic - "_get_shape" return: (1, 3, 1, 192, 384)
name: FLASH_phase_oversampling_20 - "_get_shape" return: (1, 3, 1, 231, 384)
name: FLASH_phase_resolution_75 - "_get_shape" return: (1, 3, 1, 144, 384)
name: FLASH_phase_partialfourier_6_8 - "_get_shape" return: (1, 3, 1, 144, 384)
name: FLASH_avg4 - "_get_shape" return: (4, 3, 1, 192, 384)
name: FLASH_slices3 - "_get_shape" return: (3, 3, 1, 192, 384)
name: FLASH_measures5 - "_get_shape" return: (5, 3, 1, 192, 384)
name: FLASH_GRAPPA2 - "_get_shape" return: (1, 3, 1, 191, 384)
```

Some obvervations/quenstions:

1. it looks like a frequency oversampling by a factor of 2 is set by default. I couldn't find the corresponding field in the sequence card(s) at the scanner. This is most probably the reason for the 384 values that appear in `enc matrix`.
2. I don't really understand why the recon matrix for phase resolution = 75% and partial fourier = 6/8 is the same. A reduced phase resolution should lead to reduced matrix sizes, no?
3. the k0 dimension of ktraj should NOT include the oversampling data, or?
4. the calculation of "other dim" seems to work for all tested cases

If you want to check the header information on your own, here is a code snippet to create the lists I used above (you might need to disable the shape checks in KData.from_file()):

```python
folder = Path(R'Z:\_allgemein\projects\pulseq\measurements\2023-08-15_T2MES_MRPro\FLASH')
mrd_list = list(folder.glob('*.mrd'))

kdata_list = []
for mrd in mrd_list:
    kdata_list.append(KData.from_file(filename=mrd, ktrajectory=DummyTrajectory()))

header_list = []
for kdata in kdata_list:
    header_list.append(kdata.header)
```

### PR DESCRIPTION
**Pull request by fzimmermann89**
Move  _get_shape to DummyTrajectory and ake it clear that it is only a workaround to be used until we have the first real trajectories.
Closes #42

### CODE REVIEW

**Review by schuenke — APPROVED** [review]
👍
